Assets/Scripts/Agents/AgentBase.cs

```diff
@@ -71,14 +87,12 @@ public Vector3 AccelerationInput {
     set => _accelerationInput = value;
   }
 
-  private void Awake() {
-    _rigidbody = GetComponent<Rigidbody>();
+  public float ElapsedTime {
+    get { return _elapsedTime; }
+  private
+    set { _elapsedTime = value; }
```

Comment:
nit: formatting a bit weird here with the `set` on a different line from the `private`

```csharp
public float ElapsedTime {
  get { return _elapsedTime; }
  private set { _elapsedTime = value; }
}
```